components/server/src/projects/projects-service.ts

```diff
@@ -194,6 +194,34 @@ export class ProjectsService {
         return result;
     }
 
+    async canCreateProject(currentUser: User, cloneURL: string) {
```

Comment:
I don't think this belongs to project (besides the function name).
Can we introduce an `scm-service` and move it there? The methodName could be `canInstallWebhook`

Reply:
Do you want me to externalize it to scm-service.ts? 
I'm fine with that. If there is nothing else, I'll make it happen at the end of the review, ok?